sys/phydat/phydat_json.c

```diff
@@ -84,13 +85,11 @@ size_t phydat_to_json(const phydat_t *data, size_t dim, char *buf)
             buf[pos++] = ',';
         }
         /* add unit */
-        memcpy(&buf[pos], "\"u\":\"", 5);
+        flash_memcpy(&buf[pos], TO_FLASH("\"u\":\""), 5);
```

Comment:
Isn't it implicit that the second argument of `flash_memcpy` must always be in the flash?

Reply:
Yes. The issue is that string literals are not placed in flash. Ideally a `_Generic` would be used to just force string literals to flash, but also except `__flash const char *` pointers, and fail to compile for `char *` or `const char *`.

However, there seems to be no way to actually test for string literals :/